add_flag is ambiguous with c++20

Code:
```
#include <CLI/CLI.hpp>

int main()
{
    auto app = CLI::App{"test"};
    auto flag = app.add_flag("--flag", "desc");
    return 0;
}

```

Compiling with
`g++ -std=c++20 cli11.cpp`

results in
```
cli11.cpp: In function 'int main()':
cli11.cpp:6:29: error: call of overloaded 'add_flag(const char [7], const char [5])' is ambiguous
```

CLI11 is version 2.1.2. g++ is 11.2.0.

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -102,6 +102,9 @@ jobs:
       gcc9:
         containerImage: gcc:9
         cli11.std: 17
+      gcc11:
+        containerImage: gcc:11
+        cli11.std: 20
       gcc8:
         containerImage: gcc:8
         cli11.std: 17

--- a/include/CLI/App.hpp
+++ b/include/CLI/App.hpp
@@ -799,10 +799,10 @@ class App {
 
     /// Add option for flag with integer result - defaults to allowing multiple passings, but can be forced to one
     /// if `multi_option_policy(CLI::MultiOptionPolicy::Throw)` is used.
-    template <typename T,
+    template <
+        typename T,
         enable_if_t<std::is_constructible<T, std::int64_t>::value && !std::is_const<T>::value && !is_bool<T>::value,
-                    detail::enabler> =
-                  detail::dummy>
+                    detail::enabler> = detail::dummy>
     Option *add_flag(std::string flag_name,
                      T &flag_count,  ///< A variable holding the count
                      std::string flag_description = "") {

--- a/include/CLI/App.hpp
+++ b/include/CLI/App.hpp
@@ -800,7 +800,8 @@ class App {
     /// Add option for flag with integer result - defaults to allowing multiple passings, but can be forced to one
     /// if `multi_option_policy(CLI::MultiOptionPolicy::Throw)` is used.
     template <typename T,
-              enable_if_t<std::is_constructible<T, std::int64_t>::value && !is_bool<T>::value, detail::enabler> =
+        enable_if_t<std::is_constructible<T, std::int64_t>::value && !std::is_const<T>::value && !is_bool<T>::value,
+                    detail::enabler> =
                   detail::dummy>
     Option *add_flag(std::string flag_name,
                      T &flag_count,  ///< A variable holding the count

--- a/tests/AppTest.cpp
+++ b/tests/AppTest.cpp
@@ -2310,7 +2310,7 @@ TEST_CASE_METHOD(TApp, "logFormSingleDash", "[app]") {
 }
 
 TEST_CASE("C20_compile", "simple") {
-    auto app = CLI::App{"test"};
+    CLI::App app{"test"};
     auto flag = app.add_flag("--flag", "desc");
 
     app.parse("--flag");

--- a/tests/AppTest.cpp
+++ b/tests/AppTest.cpp
@@ -2314,5 +2314,5 @@ TEST_CASE("C20_compile", "simple") {
     auto flag = app.add_flag("--flag", "desc");
 
     app.parse("--flag");
-    CHECK(static_cast<bool>(flag));
+    CHECK_FALSE(flag->empty());
 }

--- a/tests/AppTest.cpp
+++ b/tests/AppTest.cpp
@@ -2308,3 +2308,11 @@ TEST_CASE_METHOD(TApp, "logFormSingleDash", "[app]") {
     CHECK(veryverbose);
     CHECK(veryveryverbose);
 }
+
+TEST_CASE("C20_compile", "simple") {
+    auto app = CLI::App{"test"};
+    auto flag = app.add_flag("--flag", "desc");
+
+    app.parse("--flag");
+    CHECK(static_cast<bool>(flag));
+}

--- a/tests/OptionalTest.cpp
+++ b/tests/OptionalTest.cpp
@@ -239,8 +239,7 @@ TEST_CASE_METHOD(TApp, "BoostOptionalEnumTest", "[optional]") {
     auto dstring = optptr->get_default_str();
     CHECK(dstring.empty());
     run();
-    auto checkOpt = static_cast<bool>(opt);
-    CHECK(!checkOpt);
+    CHECK_FALSE(checkOpt);
 
     args = {"-v", "3"};
     run();

--- a/tests/OptionalTest.cpp
+++ b/tests/OptionalTest.cpp
@@ -239,16 +239,19 @@ TEST_CASE_METHOD(TApp, "BoostOptionalEnumTest", "[optional]") {
     auto dstring = optptr->get_default_str();
     CHECK(dstring.empty());
     run();
-    CHECK(!opt);
+    auto checkOpt = static_cast<bool>(opt);
+    CHECK(!checkOpt);
 
     args = {"-v", "3"};
     run();
-    CHECK(opt);
+    checkOpt = static_cast<bool>(opt);
+    CHECK(checkOpt);
     CHECK(*opt == eval::val3);
     opt = {};
     args = {"--val", "1"};
     run();
-    CHECK(opt);
+    checkOpt = static_cast<bool>(opt);
+    CHECK(checkOpt);
     CHECK(*opt == eval::val1);
 }
 

--- a/tests/OptionalTest.cpp
+++ b/tests/OptionalTest.cpp
@@ -239,6 +239,7 @@ TEST_CASE_METHOD(TApp, "BoostOptionalEnumTest", "[optional]") {
     auto dstring = optptr->get_default_str();
     CHECK(dstring.empty());
     run();
+    auto checkOpt = static_cast<bool>(opt);
     CHECK_FALSE(checkOpt);
 
     args = {"-v", "3"};

